support/python/src/nebula_sdk/client.py

```diff
@@ -1,23 +1,31 @@
 import os
+from typing import Container, Mapping, Optional, Text, Tuple, Union
 from urllib.parse import urljoin, urlsplit, urlunsplit
 
-from requests import Session
+from requests import PreparedRequest, Response, Session
 from requests.adapters import HTTPAdapter
 
 
 class MetadataAPIAdapter(HTTPAdapter):
 
-    def __init__(self, base_url=None):
+    def __init__(self, base_url: Optional[str] = None) -> None:
         if base_url is None:
             base_url = os.environ['METADATA_API_URL']
 
         self._base_url = base_url
 
         super(MetadataAPIAdapter, self).__init__()
 
-    def send(self, request, stream=False, timeout=None, verify=True, cert=None,
-             proxies=None):
-        (_, _, path, query, fragment) = urlsplit(request.url)
+    def send(self, request: PreparedRequest, stream: bool = False,
+             timeout: Optional[Union[
+                 float, Tuple[float, float], Tuple[float, None]]
+             ] = None,
+             verify: Union[bool, str] = True,
+             cert: Optional[Union[
+                 bytes, Text, Container[Union[bytes, Text]]]
+             ] = None,
+             proxies: Optional[Mapping[str, str]] = None) -> Response:
```

Comment:
This is horribly complicated. Can it be encapsulated in it's own type? Is that something you considered? Given where we're at right now I think we can probably save that for another day but just a thought.